test/integration/scheduler_perf/update.go

```diff
@@ -0,0 +1,166 @@
+	// Internal field of the struct used for caching the mapping.
+	cachedMapping *meta.RESTMapping
+}
+
+var _ runnableOp = &updateAny{}
+
+func (c *updateAny) isValid(allowParameterization bool) error {
+	if c.Opcode != updateAnyOpcode {
+		return fmt.Errorf("invalid opcode %q; expected %q", c.Opcode, createAnyOpcode)
+	}
+	if c.TemplatePath == "" {
+		return fmt.Errorf("TemplatePath must be set")
+	}
+	if c.UpdatePerSecond < 0 {
+		return fmt.Errorf("invalid UpdatePerSecond=%d; should be non-negative", c.UpdatePerSecond)
+	}
+	// The namespace can only be checked during later because we don't know yet
+	// whether the object is namespaced or cluster-scoped.
+	return nil
+}
+
+func (c *updateAny) collectsMetrics() bool {
+	return false
+}
+
+func (c updateAny) patchParams(w *workload) (realOp, error) {
+	if c.CountParam != "" {
+		count, err := w.Params.get(c.CountParam[1:])
+		if err != nil {
+			return nil, err
+		}
+		c.Count = ptr.To(count)
+	}
+	return &c, c.isValid(false)
+}
+
+func (c *updateAny) requiredNamespaces() []string {
+	if c.Namespace == "" {
+		return nil
+	}
+	return []string{c.Namespace}
+}
+
+func (c *updateAny) run(tCtx ktesting.TContext) {
+	count := 1
+	if c.Count != nil {
+		count = *c.Count
+	}
+
+	c.cachedMapping = nil
+
+	if c.UpdatePerSecond == 0 {
+		for index := 0; index < count; index++ {
+			err := c.update(tCtx, map[string]any{"Index": index, "Count": count})
+			if err != nil {
+				tCtx.Fatalf("Failed to update object: %w", err)
+			}
+		}
+		return
+	}
+
+	ticker := time.NewTicker(time.Second / time.Duration(c.UpdatePerSecond))
+	defer ticker.Stop()
+	for index := 0; index < count; index++ {
+		select {
+		case <-ticker.C:
+			err := c.update(tCtx, map[string]any{"Index": index, "Count": count})
+			if err != nil {
+				tCtx.Fatalf("Failed to update object: %w", err)
+			}
+		case <-tCtx.Done():
+			return
+		}
+	}
+}
+
+func (c *updateAny) update(tCtx ktesting.TContext, env map[string]any) error {
+	var obj *unstructured.Unstructured
+	if err := getSpecFromTextTemplateFile(c.TemplatePath, env, &obj); err != nil {
+		return fmt.Errorf("%s: parsing failed: %w", c.TemplatePath, err)
+	}
+
+	if c.cachedMapping == nil {
+		mapping, err := restMappingFromUnstructuredObj(tCtx, obj)
+		if err != nil {
+			return err
+		}
+		c.cachedMapping = mapping
+	}
+	resourceClient := tCtx.Dynamic().Resource(c.cachedMapping.Resource)
+
+	options := metav1.UpdateOptions{
+		// If the YAML input is invalid, then we want the
+		// apiserver to tell us via an error. This can
+		// happen because decoding into an unstructured object
+		// doesn't validate.
+		FieldValidation: "Strict",
+	}
+	if c.Namespace != "" {
+		if c.cachedMapping.Scope.Name() != meta.RESTScopeNameNamespace {
+			return fmt.Errorf("namespace %q set for %q, but %q has scope %q", c.Namespace, c.TemplatePath, c.cachedMapping.GroupVersionKind, c.cachedMapping.Scope.Name())
+		}
+		_, err := resourceClient.Namespace(c.Namespace).Update(tCtx, obj, options)
```

Comment:
Have you manually tested if this works?
I'm thinking of two complications:
- The update could fail because the ResourceVersion doesn't match the one in the server.
- There might be more fields changed that you intend, because the update template probably didn't go through defaulting. Maybe this is fine?

An alternative could be to pass patches, instead of full objects, and use `Patch`.

Reply:
`scheduler_perf` has limited dependencies, so the fields of created objects won't differ much (as nothing updates them afterwards). Most problematic can be `nodeName` field in pod, that needs to be filled correctly in updated pod spec to work. Unfortunately, as far as I know it is not possible to skip some fields on update. I tested this op also for nodes and it works well, so I think we can keep it as it is.

Reply:
ok, we can always remove it or change it for Patch if handling the `nodeName` turns out to be too complicated.